data/src/main/java/org/apache/iceberg/data/TableMigrationUtil.java

```diff
@@ -251,6 +251,6 @@ private static DataFile buildDataFile(
   }
 
   public static ExecutorService migrationService(int parallelism) {
-    return parallelism == 1 ? null : ThreadPools.newWorkerPool("table-migration", parallelism);
+    return parallelism == 1 ? null : ThreadPools.newFixedThreadPool("table-migration", parallelism);
```

Comment:
I tracked the usage. This change seems correct. But it is really unintuitive. let's at least add Javadoc to this method and document that callers are expected to shutdown the executor service. We can also add Javadoc in the `listPartition` method above and explains executor service is closed inside. Anyway, the structure/chain is a little convoluted here.